### PR TITLE
feat: dynamic public routes functionality added

### PR DESCRIPTION
**Pull request by rahulyadav139**
Deno: https://palasio-lane-store.netlify.app/

### CODE REVIEW

**Review by JowelTisso — APPROVED** [review]
* Clean code 
* Looks Good

File: src/components/listing-page/product-card/ProductCard.js [hunk 1]
```diff
@@ -36,7 +38,7 @@ const ProductCard = props => {
           <i class="far fa-heart"></i>
         </button>
       </div>
-      <Link to="/">
+      <Link to={`/product/${prodId}`}>
         <h2 class="product-title">{title.substring(0, 22) + '...'}</h2>
```
Comment (mehulsatardekar) [inline src/components/listing-page/product-card/ProductCard.js:42]: you can use template literal here

File: src/pages/ProductListingPage.js [hunk 2]
```diff
@@ -1,17 +1,54 @@
...
+import { useParams } from 'react-router-dom';
+import { useState } from 'react';
 
 const ProductListingPage = props => {
+  const [products, setProducts] = useState([]);
+  const [loading, setLoading] = useState(false);
+  const params = useParams();
+
+  const category = params.category;
+  const collection = params.collection;
```
Comment (mehulsatardekar) [inline src/pages/ProductListingPage.js:13]: you can replace this like  suppose eg : const {category, collection} =useParams();

**Review by mehulsatardekar — APPROVED** [review]
Everything looks good and the naming convention of components and state variables are self-explanatory good work @rahulyadav139

**Pull request merged**